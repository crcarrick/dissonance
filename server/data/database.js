import { Model, Sequelize } from 'sequelize';

import { channel } from './channel/channel.model';
import { message } from './message/message.model';
import { server } from './server/server.model';
import { user } from './user/user.model';

export const connectDatabase = () => {
  const sequelize = new Sequelize(
    process.env.DATABASE_NAME,
    process.env.DATABASE_USERNAME,
    process.env.DATABASE_PASSWORD,
    {
      host: 'localhost',
      dialect: 'postgres',
      logging: false,
    }
  );

  const Channel = channel({ sequelize });
  const Message = message({ sequelize });
  const Server = server({ sequelize });
  const User = user({ sequelize });

  class UserServer extends Model {}

  UserServer.init({}, { sequelize, tableName: 'users_servers' });

  const models = { Channel, Message, Server, User, UserServer };

  Object.keys(models).forEach((key) => {
    if ('associate' in models[key]) {
      models[key].associate(models);
    }
  });

  return { models, sequelize };
};
