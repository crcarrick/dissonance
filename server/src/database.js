import { merge } from 'lodash';

import knexfile from '../knexfile';

export const createDbClient = (knex, database) =>
  database
    ? knex(merge(knexfile[process.env.NODE_ENV], { connection: { database } }))
    : knex(knexfile[process.env.NODE_ENV]);
