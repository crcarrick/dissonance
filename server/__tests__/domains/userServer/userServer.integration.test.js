import { gql } from 'apollo-server';

import { createDbClient } from '@dissonance/database';

import {
  createTestClient,
  generateMockData,
  seedDatabase,
} from '@dissonance/test-utils';

const knex = jest.requireActual('knex');

describe('User Integration', () => {
  const { users, servers } = generateMockData();

  let dbClient;
  beforeAll(() => {
    dbClient = createDbClient(
      knex,
      `dissonance_test_${process.env.JEST_WORKER_ID}`
    );
  });

  afterAll(async () => {
    await dbClient.destroy();
  });

  let gqlClient;
  beforeEach(async () => {
    await seedDatabase(dbClient);

    gqlClient = createTestClient({
      context: {
        user: users[0],
      },
      dbClient,
    });
  });

  test('joins a server', async () => {
    const response = await gqlClient.mutate({
      mutation: gql`
        mutation JoinServer($input: JoinServerInput!) {
          joinServer(input: $input) {
            serverId
            userId
          }
        }
      `,
      variables: {
        input: {
          serverId: servers[0].id,
        },
      },
    });

    expect(response.data.joinServer).toMatchSnapshot();
  });
});
