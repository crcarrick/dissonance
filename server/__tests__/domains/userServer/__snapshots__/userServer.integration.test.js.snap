// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Integration joins a server 1`] = `
Object {
  "serverId": "029beee1-5599-4c4b-880d-c238b3ab3d16",
  "userId": "55e1229a-03a5-4952-a3d7-b44eb2a3bb85",
}
`;
